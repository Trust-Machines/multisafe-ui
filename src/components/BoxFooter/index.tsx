import React from 'react';

import Box from '@mui/material/Box';
import {useTheme} from '@mui/material';
import {SxProps} from '@mui/system';
import {Theme} from '@mui/material/styles';

const BoxFooter = (props: { children: React.ReactNode, sx?: SxProps<Theme> }) => {
    const theme = useTheme();

    return <Box sx={
        {
            ...{
                padding: '26px 0',
                borderTop: `2px solid ${theme.palette.divider}`,
                display: 'flex',
                alignItems: 'center',
                justifyContent: 'center'
            },
            ...props.sx
        }
    }>{props.children}</Box>
}

export default BoxFooter