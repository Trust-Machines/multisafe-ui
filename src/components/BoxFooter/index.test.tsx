import {render} from '@testing-library/react';

import BoxFooter from './index';

test('1 Default render', () => {
    const rendered = render(<BoxFooter><span>children</span></BoxFooter>);
    expect(rendered.container).toMatchSnapshot()
});

test('2 Render with custom style', () => {
    const rendered = render(<BoxFooter sx={{background: 'red'}}><span>children</span></BoxFooter>);
    expect(rendered.container).toMatchSnapshot()
});

