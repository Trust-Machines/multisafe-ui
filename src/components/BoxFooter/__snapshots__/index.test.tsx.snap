// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 Default render 1`] = `
<div>
  <div
    class="MuiBox-root css-1eq8oyx"
  >
    <span>
      children
    </span>
  </div>
</div>
`;

exports[`2 Render with custom style 1`] = `
<div>
  <div
    class="MuiBox-root css-1xbe7ur"
  >
    <span>
      children
    </span>
  </div>
</div>
`;
